### ISSUE TITLE
For screen readers, add unique descriptions to "Apply" and "program details" buttons on homepage

### ISSUE THREAD

**Issue by neel-a-mehta** [civiform/civiform#1455]
On the applicant homepage, where a list of all available programs is shown, all of the programs have identical "Apply" and "Program Details" buttons. When viewing these visually, it's clear which programs these buttons apply to, but it's less clear for screen readers.

![image](https://user-images.githubusercontent.com/77128518/123152364-5b8a6700-d419-11eb-99ef-8ed365ba21f4.png)

To differentiate the buttons, we should add screen-reader-only text that explains what the buttons are for. For instance, "Apply [for program XYZ]" or "Program details [for program XYZ]".

There are 2 accepted ways to add screen-reader-only text:

- [Put an `aria-label`](https://accessible360.com/accessible360-blog/use-aria-label-screen-reader-text/)
- Add Tailwind's ['`sr-only` class](https://tailwindcss.com/docs/screen-readers)

I believe the latter is more widely supported, FWIW.

## Original tester's report 
> Actual results:
At step 4, observe within landing page more than 1 link is labelled as ' Apply' & 'program details'
> Expected results:
each control on the page should be associated with a unique label, defining the purpose of the control. Screen reader's navigate and announce the label. duplicate labels OR multiple controls having same label will cause the user to be confused.

> ACTUAL RESULT: Multiple 'Apply' buttons are presented on Get benefits page.
> EXPECTED RESULT: 'APPLY' buttons should have a unique descriptive label. When directly accessed using form control shortcuts, the user should determine what 'APPLY' buttons are for.

**Comment by neel-a-mehta** [comment]
I'm starting to see "Continue" buttons for programs I've already started the application for. We need to add unique descriptions to these too.

**Comment by jeanniefu** [comment]
This was completed.
